backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -825,6 +837,29 @@ private fun ObjCExportCodeGenerator.generateObjCImp(
     ret(genReturnValueOnSuccess(returnType))
 }
 
+private fun ObjCExportCodeGenerator.generateExceptionTypeInfoArray(baseMethod: IrFunction): LLVMValueRef =
+        exceptionTypeInfoArrays.getOrPut(baseMethod) {
+            val throwsAnnotation = findThrowsAnnotation(baseMethod)!!
+            val types = (throwsAnnotation.getValueArgument(0) as IrVararg).elements
```

Comment:
Why those casts are safe?

Reply:
Frontend and psi2ir are supposed to guarantee this.
I will make cast failures more readable though.

Reply:
Somewhat improved.